src/cogent3/core/alignment.py

```diff
@@ -1245,15 +1246,21 @@ def write(self, filename=None, format=None, **kwargs):
         if filename is None:
             raise DataError("no filename specified")
 
+        suffix, cmp_suffix = get_format_suffixes(filename)
+        if format is None and suffix:
+            format = suffix
+
+        if format == "json":
+            f = open_(filename, "wt")
```

Comment:
using the context style and `atomic_write`, change this to:
```python
with atomic_write(filename mode="wt") as f:
    f.write(self.to_json())
```
then delete the `f.close()` line